go/bin/opentelemetry/main.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"time"
+
+	"github.com/dylibso/observe-sdk/go/adapter/opentelemetry"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// we only need to create and start once per instance of our host app
+	conf := &opentelemetry.OTelConfig{
+		ServiceName:        "golang",
+		EmitTracesInterval: time.Second * 1,
+		TraceBatchMax:      100,
+		Endpoint:           "localhost:4318",
+		Protocol:           opentelemetry.HTTP,
+		AllowInsecure:      true, // for localhost in dev via http
+	}
+	adapter := opentelemetry.NewOTelAdapter(conf)
+	defer adapter.StopWithContext(ctx, true)
```

Comment:
this is a new method on the Adapter, since `Stop` is part of the `AdapterBase` interface and we can't override it with a new signature, since we _have to_ implement this interface. 

So `Stop` also calls the underlying OpenTelemetry package's Stop on their client, but `StopWithContext` allows the end user to capture the error and pass a correlated context to the function here.

Reply:
(its unique to the OTelAdapter, not needed on any others)

Reply:
:+1, not in this PR , but maybe we can converge on `StopWithContext` and nix `Stop`

Reply:
yea that's a good idea. there's no harm in taking the `ctx` in the existing `Stop` func and just supporting this new need in the higher level interface